### PR TITLE
DAOS-10912 test: Verify EC space consumption

### PR DESCRIPTION
**Pull request by daltonbohning**
Test-tag: pr EcodSpaceUsage HarnessUnitTest storage_tiers PoolCreateAllHwTests VmdLedStatus,-manual
Skip-unit-tests: true
Skip-fault-injection-test: true

- Added ftest/erasurecode/space_usage
  - Verify space consumption of EC object classes is balanced within 5%
- Added dmg_utils.py:pool_query_targets()
- Added TestPool.query_targets()
- Added TestPool.get_space_per_target()
- Added ftest/utils/data_utils.py
  - Misc utilities related to data manipulation
- Remove yaml_utils.py:find_values in favor of data_utils.py:dict_extract_values
- Added ftest/harness/unit
  - Harness unit tests for utilities
  - 
Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by daltonbohning** [comment]
Existing test failures in https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-11463/14/tests/
- FTEST_daos_test.DAOS_Rebuild_EC
  - https://daosio.atlassian.net/browse/DAOS-12709

File: src/tests/ftest/erasurecode/space_usage.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        self.log.info('Max used NVMe difference from mean: %.2f%%', max_nvme_diff * 100)
+
+        # Verify the correct number of targets were used
+        # Don't enforce SCM, since little metadata is used
+        expected_targets = calculate_ec_targets_used(self.ior_cmd.dfs_oclass.value, total_targets)
+        if len(nvme_used) != expected_targets:
+            self.fail('Incorrect number of targets used!')
+
+        # Verify space usage across targets is balanced
+        # Don't enforce SCM, since little metadata is used
```
Comment (shimizukko) [inline src/tests/ftest/erasurecode/space_usage.py:117]: Do you need this comment line? Looks like a duplicate of line 117.

**Review by shimizukko — DISMISSED** [review]

File: src/tests/ftest/erasurecode/space_usage.py (hunk 1 above)
Comment (daltonbohning) [reply src/tests/ftest/erasurecode/space_usage.py:117]: It is a duplicate, but I wanted to be clear that we aren't enforcing SCM

**Review by phender — DISMISSED** [review]

**Review by phender — APPROVED** [review]

**Review by shimizukko — APPROVED** [review]

**Comment by daltonbohning** [comment]
Tests passed with
`Test-tag: pr EcodSpaceUsage HarnessUnitTest storage_tiers PoolCreateAllHwTests VmdLedStatus,-manual DfuseMUPerms ior_intercept DfuseCachingCheck`
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-11463/15/tests

**Pull request merged**